sshkey.c

```diff
@@ -3406,12 +3524,18 @@ sshkey_parse_private_pem_fileblob(struct sshbuf *blob, int type,
 			r = SSH_ERR_ALLOC_FAIL;
 			goto out;
 		}
-		prv->rsa = EVP_PKEY_get1_RSA(pk);
+		prv->pkey = pk;
+		pk = NULL;
 		prv->type = KEY_RSA;
+		if ((rsa = EVP_PKEY_get1_RSA(prv->pkey)) == NULL) {
+			r = SSH_ERR_LIBCRYPTO_ERROR;
+			goto out;
+		}
 #ifdef DEBUG_PK
-		RSA_print_fp(stderr, prv->rsa, 8);
+		RSA_print_fp(stderr, rsa, 8);
 #endif
-		if (RSA_blinding_on(prv->rsa, NULL) != 1) {
+		if (RSA_blinding_on(rsa, NULL) != 1 ||
+		    !EVP_PKEY_set1_RSA(prv->pkey, rsa)) {
```

Comment:
is it ok to do this? rsa  = get1_rsa(pk); set1_rsa(pk, rsa); ?

Reply:
since rsa is already the rsa in pk?

Reply:
or should rsa=get0_rsa(pk); enable_blindung(rsa) be used?

Reply:
Blinding is ON in modern OpenSSL by default, AFAIK